modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,59 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analsyes_object_array
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analsyes_object_array);
+  Description:
+    Reset jobs for the specified $input_id pattern and $analsyis_object_array. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, @analyses_list) = @_;
+
+    #Convert MySQL wildcard pattern to Perl wildcard pattern
+    if ($input_id_pattern =~ /\A[^%]/) {
+        $input_id_pattern = '^'.$input_id_pattern;
+    }
+    if ($input_id_pattern =~ /[^%]\z/) {
+        $input_id_pattern = $input_id_pattern.'$';
+    }
+    $input_id_pattern =~ s/\%/\.*/g;
+
+    my $analysis;
+    my $jobs = $self->fetch_all_by_analysis_id_status(@analyses_list);
+    foreach my $job (@$jobs) {
+         my $input_id = $job->input_id;
+         $input_id_pattern = qr/$input_id_pattern/;
+         if ($input_id =~ /$input_id_pattern/) {
+                my $job_status = $job->status();
+                my $job_info = $job->toString;
+                if($job_status =~ /CLAIMED|SEMAPHORED/) {
+                       die "$job_info is $job_status, cannot reset";
+                }
+
+                if ($job_status ne 'READY') {
```

Comment:
Since you have removed the line with `ALL_STATUSES_OF_RUNNING_JOBS`, this will now catch jobs that are running. These should _not_ be reset.
The test you had in  previous commit: https://github.com/Ensembl/ensembl-hive/blob/6f07a94bfe737c5911bf24dab06c13588af8e153/modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm#L614-L616 was fine and should be reinstated, but now taking into account my new comment: should it _die_ or _skip_ ?